src/web/app/components/question-types/question-edit-details-form/question-edit-details-form.component.ts

```diff
@@ -38,4 +38,33 @@ export abstract class QuestionEditDetailsFormComponent<D extends FeedbackQuestio
   triggerModelChangeBatch(obj: Partial<D>): void {
     this.detailsChange.emit({ ...this.model, ...obj });
   }
+
+  onPointsInput(event: KeyboardEvent): void {
+    const { key } = event;
+    const isBackspace = key === 'Backspace';
+    const isDigit = /[0-9]/.test(key);
+    if (!isBackspace && !isDigit) {
+      event.preventDefault();
+    }
+  }
+
+  onPaste(event: ClipboardEvent): void {
+    const { clipboardData } = event;
+    if (clipboardData == null) {
+      return;
+    }
+    const pastedText = clipboardData.getData('text');
+    const isDigit = /^\d+$/.test(pastedText);
+    if (!isDigit) {
+      event.preventDefault();
+    }
+  }
+
+  restrictPointsLength(event : InputEvent, field: keyof D) : void {
+    const target : HTMLInputElement = event.target as HTMLInputElement;
+    if (target.value != null && target.value.length > 9) {
+      target.value = target.value.substring(0, 9);
+      this.triggerModelChange(field, parseInt(target.value, 10) as any);
+    }
+  }
```

Comment:
`restrictFloatInputLength` can follow the exact same implementation as `restrictPointsLength` method, except: `parseInt(target.value, 10)` can be changed to `parseFloat(target.value)`